_posts/2020-06-07-higher-order-hydras.markdown

```diff
@@ -0,0 +1,491 @@
+
+<div>
+
+Before we proceed to make stronger hydras, I should show a method to draw nested hydras.
+It would be inconvenient to try and embed a diagram of a hydra within a single node, and possibly even do that recursively.
+Instead, I propose drawing the label to the left of the node, connected using a sideways edge.
+
+![Two diagrams of the same Buchholz hydra, side by side. One of them draws the labels on the node, while the other draws the label to the left of the node with an edge between them.](/assets/hydras/left-labels.png)
+
+This looks unnecessary and silly when the labels are small ordinals like this, but when we use hydras as labels, this becomes much more practical.
+
+</div>
+
+# Externalize and close
+{: #externalize-close}
+
+{% include collapser.markdown %}
+
+<div>
+
+"Externalize" and "close" are some important basic transformation types in our toolkit of higher order transformations on hydra constructions.
+"Externalize" means to shift information from the labels (internal) toward the children (external), with the end goal of putting an upper bound on the labels without changing the strength of the construction.
+"Close" means to take a construction where labels are upper-bounded, and extend it to support arbitrary hydras as labels, thus taking the closure of the construction under using hydras as labels, and making the construction stronger.
+These form a natural pair.
+I call these transformation types because there is not just one way to externalize or close, and different possible transformations may differ in strength.
+
+Amiko hydras can be derived from Buchholz hydras with a close step, a somewhat strong externalize step, then a close step.
+The interesting step here is the externalize.
+I'll walk you through converting the example hydra from above, and then discuss the strength of the system.
+
+![Diagram of a Buchholz hydra with the labels drawn to the left instead of on the nodes.](/assets/hydras/externalize-0.png)
+
+We're first going to mark all existing edges with label \\\(0\\\).
+
+![Diagram of a Buchholz hydra with the labels drawn to the left instead of on the nodes, and edges labelled with 0.](/assets/hydras/externalize-1.png)
+
+We're then going to move the current label into a child node, with edge labelled \\\(1\\\).
+
+![Diagram of a Buchholz hydra with the former labels drawn as children with edge label 1, and original edges labelled with 0.](/assets/hydras/externalize-2.png)
+
+All original labels have now been evicted.
+We transfer these edge labels into the now unlabelled children.
+The root could be considered to have a special fake label, or no label.
+
+![Diagram of a Buchholz hydra with the former labels drawn as children with label 1, and original nodes labelled 0.](/assets/hydras/externalize-3.png)
+
+We then expand these former labels into the appropriate hydras, except that the immediate children of that root has label \\\(1\\\), and then fuse those with their parents.
```

Comment:
I find this sentence a bit confusing. Are you saying "We then expand these former labels into the appropriate hydras, except that if the immediate children of that root have label 1, we fuse those with their parents"?

Reply:
Oh, I think this is a legitimate mistake. I'll go see what it should be. Thanks for pointing it out.

Reply:
Actually, no, this means exactly what it says. Second clause binds to the first, "then" separates them:

We then [expand these former labels into the appropriate hydras, except that the immediate children of that root has label \\\(1\\\)], and then [fuse those with their parents].

I agree it's confusing though. I'll think about how to better convey the same meaning.

Reply:
No problem! I see where the clauses are now though too 👍